tests/lib/tools/tests.env

```diff
@@ -0,0 +1,126 @@
+#!/bin/bash
+
+show_help() {
+    echo "usage: tests.env start <ENV_NAME>"
+    echo "       tests.env is-set <ENV_NAME> <VAR>"
+    echo "       tests.env get <ENV_NAME> <VAR>"
+    echo "       tests.env set <ENV_NAME> <VAR> <VAL>"
+}
+
+start() {
+    local NAME=$1
+    if [ -z "$NAME" ]; then
+        echo "tests.env: name for the env file is required"
+        exit 1
+    fi
+
+    if [ -f "$RUNTIME_STATE_PATH/$NAME.env" ]; then
+        echo "tests.env: env file $RUNTIME_STATE_PATH/$NAME.env already exists, deleting..."
+        rm -f "$RUNTIME_STATE_PATH/$NAME.env"
+    fi
+    mkdir -p "$RUNTIME_STATE_PATH"
+    touch "$RUNTIME_STATE_PATH/$NAME.env"
+}
+
+is_set() {
+    local NAME=$1
+    local VAR=$2
+
+    if [ -z "$NAME" ]; then
+        echo "tests.env: name for the env file is required"
+        exit 1
+    fi
+    if [ -z "$VAR" ]; then
+        echo "tests.env: variable to check in env file is required"
+        exit 1
+    fi
+
+    if [ ! -f "$RUNTIME_STATE_PATH/$NAME.env" ]; then
+        echo "tests.env: env file $RUNTIME_STATE_PATH/$NAME.env does not exist"
+        exit 1
+    fi
+
+    grep -Eq "^${VAR}=" "$RUNTIME_STATE_PATH/$NAME.env"
+}
+
+get() {
+    local NAME=$1
+    local VAR=$2
+
+    if [ -z "$NAME" ]; then
+        echo "tests.env: name for the env file is required"
+        exit 1
+    fi
+    if [ -z "$VAR" ]; then
+        echo "tests.env: variable to check in env file is required"
+        exit 1
+    fi
+
+    if [ ! -f "$RUNTIME_STATE_PATH/$NAME.env" ]; then
+        echo "tests.env: env file $RUNTIME_STATE_PATH/$NAME.env does not exist"
+        exit 1
+    fi
+
+    if is_set "$NAME" "$VAR"; then
+        grep -E "^${VAR}=" "$RUNTIME_STATE_PATH/$NAME.env" | cut -d "=" -f2-
+    fi
+}
+
+set() {
+    local NAME=$1
+    local VAR=$2
+    local VAL=$3
+
+    if [ -z "$NAME" ]; then
+        echo "tests.env: name for the env file is required"
+        exit 1
+    fi
+    if [ -z "$VAR" ]; then
+        echo "tests.env: variable to set in env file is required"
+        exit 1
+    fi
+
+    if [ ! -f "$RUNTIME_STATE_PATH/$NAME.env" ]; then
+        echo "tests.env: env file $RUNTIME_STATE_PATH/$NAME.env does not exist"
+        exit 1
+    fi
+
+    if is_set "$NAME" "$VAR"; then
+        sed -i "/^${VAR}=/d" "$RUNTIME_STATE_PATH/$NAME.env"
+    fi
+    echo "${VAR}=${VAL}" >> "$RUNTIME_STATE_PATH/$NAME.env"
+
+}
+
+main() {
+    if [ $# -eq 0 ]; then
+        show_help
+        exit 0
+    fi
+
+    local subcommand="$1"
+    local action=
+    while [ $# -gt 0 ]; do
+        case "$1" in
+            -h|--help)
+                show_help
+                exit 0
+                ;;
+            *)
+                action=$(echo "$subcommand" | tr '-' '_')
+                shift
+                break
+                ;;
+        esac
+    done
+
+    if [ -z "$(declare -f "$action")" ]; then
+        echo "tests.env: no such command: $subcommand"
```

Comment:
Nitpick, more general than this specific review:

`grep -r "no such command"`  shows that similar subcommand selection messages does not seem consistent, in that some report `action` which is reformatted from "the-sub-command" to "the_sub_command" (e.g. lib/external/snapd-testing-tools/remote/remote.wait-for) and some like this one reports the `subcommand` as passed in format "the-sub-command".

Reply:
Sure, I'll create a new pr to unify that.
Thanks for pointing this.